corehq/apps/domain/views/settings.py

```diff
@@ -581,16 +583,32 @@ def post(self, request, *args, **kwargs):
             except ValidationError as e:
                 messages.error(request, _(e.message))
                 return HttpResponseRedirect(request.path_info)
+            uidb64 = kwargs.get('uidb64')
+            uid = urlsafe_base64_decode(uidb64)
+            user = User.objects.get(pk=uid)
+            couch_user = CouchUser.from_django_user(user)
+            clear_login_attempts(couch_user)
```

Comment:
can you explain what this bug was? "Fixed a small bug where login attempts were being cleared even if the password reset wasn't successful" 

I would think all of this should be done after `response = super().post(request, *args, **kwargs)` in case something fails in the parent's `post`

Reply:
cool, and i may just be misunderstanding what's going on here and i don't mean to harp on this, should `response = super().post(request, *args, **kwargs)` happen before `clear_login_attempts` and sending the success email?

My main worry is that we already send the "success" email but then a failure occurs during the super().post that means the password isn't actually set.

Reply:
no worries, moved the call back to before the post-success actions [f9cb8ad](https://github.com/dimagi/commcare-hq/pull/36711/commits/f9cb8ade32c550433854ca28539700281f229cb1)